pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,22 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// TaintNotToleratedWithFilter checks if the given tolerations tolerates
+// all the filtered taints, and returns the first taint without a toleration
+func TaintNotToleratedWithFilter(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) *v1.Taint {
```

Comment:
I am not sure about the name of the function, I don't have better suggestions, so will leave it to core api approvers to suggest a new name (or confirm this one).

Reply:
changed this